ENH / FF: iohub hdf5 builder integration and saveEventReport updates

isolver opened the pull request:
FF: Remove iohub cacheMessageEvent and sendMessageEvents calls from Builder boilerplate
ENH: Added psychopy .csv trial data and hdf5 file merge support to iohub saveEventReports function.  

TODO: Update / check demo's.

isolver committed:
ENH: initial pass at merging .csv data with .hdf5 events

ENH: initial pass at merging .csv data with .hdf5 events

isolver committed:
FF: added psychopyDataFileColumns arg to saveEventReport

FF: added psychopyDataFileColumns arg to saveEventReport

isolver committed:
FF: Remove iohub cacheMessageEvent and sendMessageEvents from Builder boilerplate

FF: Remove iohub cacheMessageEvent and sendMessageEvents from Builder boilerplate

isolver committed:
Merge remote-tracking branch 'upstream/dev' into nf_iohub_saveEventReport_with_builder_data_file

isolver committed:
more work on iohub hdf5 saveEventReport feature

isolver committed:
iohub saveEventReport: Optionally select event grouping columns

iohub saveEventReport: Optionally select event grouping columns when using psychopy .csv

isolver committed:
Merge remote-tracking branch 'upstream/dev' into nf_iohub_saveEventReport_with_builder_data_file

TEParsons commented on psychopy/experiment/components/_base.py:
```diff
@@ -839,31 +839,6 @@ def writeFrameCodeJS(self, buff):
             buff.setIndentLevel(-1, relative=True)  # to exit the if block
             buff.writeIndented("}\n")
 
-    def writeRoutineEndCode(self, buff):
```
I'm guessing you're removing this as sending events to ioHub is now done in the function which saves to PsychoPy?

isolver committed:
FF: finished saveEventReport hdf5 cv table <-> event table merge

FF: finished saveEventReport hdf5 condition variables table <-> event table merge functionality

isolver committed:
Merge remote-tracking branch 'upstream/dev' into nf_iohub_saveEventReport_with_builder_data_file

isolver committed:
ENH: Added  iohub.constants.EYE_SAMPLE_TYPES list

ENH: Added  iohub.constants.EYE_SAMPLE_TYPES list
ENH: Added getAvailableEyeSampleTypes method to iohub hdf5 reader class (ExperimentDataAccessUtility class)
